--- a/inc/internal_gcal.h
+++ b/inc/internal_gcal.h
@@ -50,22 +50,22 @@ typedef xmlDoc dom_document;
 
 static const char GCAL_DELIMITER[] = "%40";
 static const char GCAL_URL[] = "https://www.google.com/accounts/ClientLogin";
-static const char GCAL_LIST[] = "http://www.google.com/calendar/feeds/"
+static const char GCAL_LIST[] = "https://www.google.com/calendar/feeds/"
 	"default/allcalendars/full";
 /* Google calendar URL for posting new events */
-static const char GCAL_EDIT_URL[] = "http://www.google.com/calendar/feeds"
+static const char GCAL_EDIT_URL[] = "https://www.google.com/calendar/feeds"
 	"/default/private/full";
 /* Google contacts URL for posting new contacts */
-static const char GCONTACT_EDIT_START[] = "http://www.google.com/m8/feeds/"
+static const char GCONTACT_EDIT_START[] = "https://www.google.com/m8/feeds/"
 	"contacts/";
 static const char GCONTACT_EDIT_END[] = "/full";
 
 /* Google calendar query URL */
-static const char GCAL_EVENT_START[] = "http://www.google.com/calendar/feeds/";
+static const char GCAL_EVENT_START[] = "https://www.google.com/calendar/feeds/";
 static const char GCAL_EVENT_END[] = "/private/full";
 
 /* Google contact query URL */
-static const char GCONTACT_START[] = "http://www.google.com/m8/feeds/contacts/";
+static const char GCONTACT_START[] = "https://www.google.com/m8/feeds/contacts/";
 static const char GCONTACT_END[] = "/full";
 
 /* Google 'pages' results in a range pages of 25 entries. But for downloading
