From 431911e8847aa67aea3607debd9c17a617d9a9b9 Mon Sep 17 00:00:00 2001
From: Jan Grulich <jgrulich@redhat.com>
Date: Fri, 16 Jan 2015 11:26:16 +0100
Subject: [PATCH] Checking for security in wireless setting won't work for NM
 1.0.0+

---
 utils.cpp | 40 +++++++++++++++++++---------------------
 1 file changed, 19 insertions(+), 21 deletions(-)

diff --git a/utils.cpp b/utils.cpp
index 71e34db..82339fe 100644
--- a/utils.cpp
+++ b/utils.cpp
@@ -407,29 +407,27 @@ bool NetworkManager::Utils::wpaPskIsValid(const QString &psk)
 
 NetworkManager::Utils::WirelessSecurityType NetworkManager::Utils::securityTypeFromConnectionSetting(const NetworkManager::ConnectionSettings::Ptr& settings)
 {
-    NetworkManager::WirelessSetting::Ptr wifiSetting = settings->setting(Setting::Wireless).dynamicCast<WirelessSetting>();
-    if (!wifiSetting->security().isEmpty()) {
-        NetworkManager::WirelessSecuritySetting::Ptr wifiSecuritySetting = settings->setting(Setting::WirelessSecurity).dynamicCast<WirelessSecuritySetting>();
-        if (wifiSecuritySetting->keyMgmt() == WirelessSecuritySetting::Wep) {
-            return Utils::StaticWep;
-        } else if (wifiSecuritySetting->keyMgmt() == WirelessSecuritySetting::Ieee8021x) {
-            if (wifiSecuritySetting->authAlg() == WirelessSecuritySetting::Leap) {
-                return Utils::Leap;
-            } else {
-                return Utils::DynamicWep;
-            }
-        } else if (wifiSecuritySetting->keyMgmt() == WirelessSecuritySetting::WpaPsk) {
-            if (wifiSecuritySetting->proto().contains(WirelessSecuritySetting::Rsn) && !wifiSecuritySetting->proto().contains(WirelessSecuritySetting::Wpa)) {
-                return Utils::Utils::Wpa2Psk;
-            }
-            return Utils::WpaPsk;
-        } else if (wifiSecuritySetting->keyMgmt() == WirelessSecuritySetting::WpaEap) {
-            if (wifiSecuritySetting->proto().contains(WirelessSecuritySetting::Rsn) && !wifiSecuritySetting->proto().contains(WirelessSecuritySetting::Wpa)) {
-                return Utils::Wpa2Eap;
-            }
-            return Utils::Utils::WpaEap;
+    NetworkManager::WirelessSecuritySetting::Ptr wifiSecuritySetting = settings->setting(Setting::WirelessSecurity).dynamicCast<WirelessSecuritySetting>();
+    if (wifiSecuritySetting->keyMgmt() == WirelessSecuritySetting::Wep) {
+        return Utils::StaticWep;
+    } else if (wifiSecuritySetting->keyMgmt() == WirelessSecuritySetting::Ieee8021x) {
+        if (wifiSecuritySetting->authAlg() == WirelessSecuritySetting::Leap) {
+            return Utils::Leap;
+        } else {
+            return Utils::DynamicWep;
+        }
+    } else if (wifiSecuritySetting->keyMgmt() == WirelessSecuritySetting::WpaPsk) {
+        if (wifiSecuritySetting->proto().contains(WirelessSecuritySetting::Rsn) && !wifiSecuritySetting->proto().contains(WirelessSecuritySetting::Wpa)) {
+            return Utils::Utils::Wpa2Psk;
         }
+        return Utils::WpaPsk;
+    } else if (wifiSecuritySetting->keyMgmt() == WirelessSecuritySetting::WpaEap) {
+        if (wifiSecuritySetting->proto().contains(WirelessSecuritySetting::Rsn) && !wifiSecuritySetting->proto().contains(WirelessSecuritySetting::Wpa)) {
+            return Utils::Wpa2Eap;
+        }
+        return Utils::Utils::WpaEap;
     }
+
     return Utils::None;
 }
 
-- 
2.2.2

